h5py/_hl/files.py

```diff
@@ -39,6 +39,47 @@
 libver_dict_r = dict((y, x) for x, y in six.iteritems(libver_dict))
 
 
+def _set_fapl_mpio(plist, **kwargs):
+    kwds.setdefault('info', mpi4py.MPI.Info())
+    plist.set_fapl_mpio(**kwargs)
+
+
+drivers = {
+    'sec2': lambda plist, **kwargs: plist.set_fapl_sec2(**kwargs),
+    'stdio': lambda plist, **kwargs: plist.set_fapl_stdio(**kwargs),
+    'core': lambda plist, **kwargs: plist.set_fapl_core(**kwargs),
+    'family': lambda plist, **kwargs: plist.set_fapl_family(
+        memb_fapl=plist.copy(),
+        **kwargs,
+    ),
+    'mpio': _set_fapl_mpio,
+}
+
+
+def register_driver(name, set_fapl):
+    """Register a custom driver.
+
+    Parameters
+    ----------
+    name : str
+        The name of the driver.
+    set_fapl : callable[PropFAID, **kwargs, NoneType]
```

Comment:
I'm confused by the NoneType, why is it there?

Reply:
This was saying that the signature is PropFAID, then keyword arguments, with a return type of NoneType. I don't see a lot of examples for documenting callbacks so this is what I have used. I would be happy to change this to whatever you prefer. One option would just be "callable" and then describe the signature below.

Reply:
Is `callable[PropFAID, **kwargs] -> NoneType` a thing, or is my knowledge of the new typing syntax just wrong?

Reply:
I don't think typing uses this at all, but that is more clear than what I wrote.